lib/transport.dart

```diff
@@ -210,6 +210,11 @@ abstract class TransportStream {
   /// A sink for writing data and/or headers to the remote end.
   StreamSink<StreamMessage> get outgoingMessages;
 
+  /// Set the cancellation handler on this stream.
+  ///
+  /// The handler will be called if the stream receives an RST_STREAM frame.
+  void onCancel(void value(int errorCode));
```

Comment:
Maybe we should also rename it: The term "cancelling" is gRPC terminology, here we could use http/2 terminology "reset" / "terminate".